### ISSUE TITLE
Mod crashes game at startup

### ISSUE THREAD

**Issue by RedBloodCat** [egladil/oni-nubrion#6]
When i activated this mod the game crashes on startup everytime, but when i deactivate its works. 

Error code i get when starting up with mod;
  at UnityEngine.Debug.LogError (System.Object message) [0x00000] in <a35d771e78bd4d75a6f3aedeaad4d1ed>:0 
  at Debug.LogError (System.Object obj) [0x00000] in <e3c6a70dbf804766b5f7ed03fef8497e>:0 
  at Egladil.Log.Error (System.String str) [0x00000] in <03574ca38dd1480897c99fba3d74b26a>:0 
  at Egladil.FusionReactor.InitRadiation () [0x00000] in <03574ca38dd1480897c99fba3d74b26a>:0 
  at Egladil.FusionReactorConfig.CreateBuildingDef () [0x00000] in <03574ca38dd1480897c99fba3d74b26a>:0 
  at BuildingConfigManager.RegisterBuilding (IBuildingConfig config) [0x00000] in <e38a27e719b642cba3ce0b7a77b47055>:0 
  at GeneratedBuildings.LoadGeneratedBuildings_Patch8 (System.Collections.Generic.List`1[T] ) [0x00000] in <e38a27e719b642cba3ce0b7a77b47055>:0 
  at LegacyModMain.LoadBuildings (System.Collections.Generic.List`1[T] types) [0x00000] in <e38a27e719b642cba3ce0b7a77b47055>:0 
  at LegacyModMain.Load () [0x00000] in <e38a27e719b642cba3ce0b7a77b47055>:0 
  at Assets.CreatePrefabs () [0x00000] in <e38a27e719b642cba3ce0b7a77b47055>:0 
  at Assets.OnPrefabInit () [0x00000] in <e38a27e719b642cba3ce0b7a77b47055>:0 
  at KMonoBehaviour.InitializeComponent () [0x00000] in <e3c6a70dbf804766b5f7ed03fef8497e>:0 
  at KMonoBehaviour.Awake () [0x00000] in <e3c6a70dbf804766b5f7ed03fef8497e>:0 
  at UnityEngine.Object.Internal_InstantiateSingleWithParent_Injected (UnityEngine.Object data, UnityEngine.Transform parent, UnityEngine.Vector3& pos, UnityEngine.Quaternion& rot) [0x00000] in <a35d771e78bd4d75a6f3aedeaad4d1ed>:0 
  at UnityEngine.Object.Internal_InstantiateSingleWithParent (UnityEngine.Object data, UnityEngine.Transform parent, UnityEngine.Vector3 pos, UnityEngine.Quaternion rot) [0x00000] in <a35d771e78bd4d75a6f3aedeaad4d1ed>:0 
  at UnityEngine.Object.Instantiate (UnityEngine.Object original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.Transform parent) [0x00000] in <a35d771e78bd4d75a6f3aedeaad4d1ed>:0 
  at UnityEngine.Object.Instantiate[T] (T original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.Transform parent) [0x00000] in <a35d771e78bd4d75a6f3aedeaad4d1ed>:0 
  at Util.KInstantiate (UnityEngine.GameObject original, UnityEngine.Vector3 position, UnityEngine.Quaternion rotation, UnityEngine.GameObject parent, System.String name, System.Boolean initialize_id, System.Int32 gameLayer) [0x00000] in <e3c6a70dbf804766b5f7ed03fef8497e>:0 
  at Util.KInstantiate (UnityEngine.GameObject original, UnityEngine.GameObject parent, System.String name) [0x00000] in <e3c6a70dbf804766b5f7ed03fef8497e>:0 
  at LaunchInitializer.Update () [0x00000] in <e38a27e719b642cba3ce0b7a77b47055>:0

**Comment by TroublesomeCorvid** [comment]
This looks similar to an issue I'm having with both the Fusion Reactor and the Radiation mods.  Both of them together without any other mods active, or even by themselves, will crash the game on start-up for some reason.  I'm including a txt file with the error messages for the following cases:

- Fusion Reactor with other mods active
- Radiation with other mods active
- Fusion Reactor and Radiation without other mods active
- Fusion Reactor without other mods active
- Radiation without other mods active

Hope this helps.
[error messages.txt](https://github.com/egladil/oni-mods/files/4888428/error.messages.txt)

**Comment by Fatmice** [comment]
Should mention that it is with AP-419840-D

**Comment by Fatmice** [comment]
System.TypeInitializationException: The type initializer for 'Egladil.RadiationGerms' threw an exception. ---> System.MissingFieldException: Field 'DiseaseVisualization/Info.overlayColour' not found.
   --- End of inner exception stack trace ---
  at (wrapper managed-to-native) System.Object.__icall_wrapper_mono_generic_class_init(intptr)
  at Egladil.RadiationGerms+Diseases_Constructor_Patch.Postfix (Database.Diseases __instance) [0x00001] in <ded48b2c175e458e97244cfeace77265>:0 
  at (wrapper dynamic-method) Database.Diseases..ctor_Patch1(object,ResourceSet)
  at Db.Initialize () [0x0016c] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at Db.Get () [0x0001c] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at BuildingDef.PostProcess () [0x000b3] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at BuildingConfigManager.RegisterBuilding (IBuildingConfig config) [0x0012b] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at GeneratedBuildings.LoadGeneratedBuildings (System.Collections.Generic.List`1[T] types) [0x00072] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at LegacyModMain.LoadBuildings (System.Collections.Generic.List`1[T] types) [0x00050] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at LegacyModMain.Load () [0x00042] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at Assets.CreatePrefabs () [0x00000] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at Assets.OnPrefabInit () [0x0031c] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at KMonoBehaviour.InitializeComponent () [0x00068] in <29173bb5b2e54d1e9a68fc8be8c6904a>:0 
Error in GameAssets(Clone).Assets.OnPrefabInit 
System.TypeInitializationException: The type initializer for 'Egladil.RadiationGerms' threw an exception. ---> System.MissingFieldException: Field 'DiseaseVisualization/Info.overlayColour' not found.
   --- End of inner exception stack trace ---
  at (wrapper managed-to-native) System.Object.__icall_wrapper_mono_generic_class_init(intptr)
  at Egladil.RadiationGerms+Diseases_Constructor_Patch.Postfix (Database.Diseases __instance) [0x00001] in <ded48b2c175e458e97244cfeace77265>:0 
  at (wrapper dynamic-method) Database.Diseases..ctor_Patch1(object,ResourceSet)
  at Db.Initialize () [0x0016c] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at Db.Get () [0x0001c] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at BuildingDef.PostProcess () [0x000b3] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at BuildingConfigManager.RegisterBuilding (IBuildingConfig config) [0x0012b] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at GeneratedBuildings.LoadGeneratedBuildings (System.Collections.Generic.List`1[T] types) [0x00072] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at LegacyModMain.LoadBuildings (System.Collections.Generic.List`1[T] types) [0x00050] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at LegacyModMain.Load () [0x00042] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at Assets.CreatePrefabs () [0x00000] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at Assets.OnPrefabInit () [0x0031c] in <35509e1c59af46fc9883c0f96c2746f7>:0 
  at KMonoBehaviour.InitializeComponent () [0x00068] in <29173bb5b2e54d1e9a68fc8be8c6904a>:0

**Comment by geonmo** [comment]
@Fatmice @RedBloodCat @TroublesomeCorvid Once I checked the code, I found that there was no problem in Fusion Reactor mod. In the Radiation mod, the contents related to color were changed from color code to color name, so I modified it.

Please go to the [link](https://drive.google.com/drive/folders/1NUNTqaenH2vEhWDMaYcyyRf6MwUSsBWD) to download the radiation mod and install it manually to check its operation.  

I wanted to use cyan as the existing color setting, but I chose the color of "zombie spore" because I was not familiar with the color set.

I think that a steam version of the "fusion reactor" mod is working. So, you need to check the "radiation" mod only.

Regards,

**Comment by Fatmice** [comment]
@geonmo I tested Radiation, FusionReactor, GigawattWire, and Hypercoolant in an old save and everything seems to be running fine. Thank you for your help. There is indeed no problem with the FusionReactor mod but only that since Radiation mod failed to load so the FusionReactor mod also failed to load.

Downloaded your versions
Rename
Radiation.zip > 1953157233.zip
FusionReactor.zip > 1953154411.zip
Extracted GigawattWire.zip and rezip the contents into 1984954968.zip

sha256sums (GNU coreutils) 8.31
6c9cb4ce7750db02c578ebf3ae7a27f64a315289a4f9c75e2278b539db39f5c2  FusionReactor/FusionReactor.dll
82d9a827cb5a20c79b89fdae68a5ba75005efcc5f08e5ddca382bc5c3cceb3b9  GigawattWire/GigawattWire.dll
7a0be59468323d021e141aa3e971fe3e6292c3781750b4e0a1ce150d667022f3  Radiation/Radiation.dll

Then I extracted and overwritten the contents of those corresponding folders in Oxygen Not Included\mods\Steam because I have ModUpdater and ModManager. I did not force update.

Attached are the renamed zips
[1953154411.zip](https://github.com/egladil/oni-mods/files/4904679/1953154411.zip)
[1953157233.zip](https://github.com/egladil/oni-mods/files/4904680/1953157233.zip)
[1984954968.zip](https://github.com/egladil/oni-mods/files/4904681/1984954968.zip)

sha256sums (GNU coreutils) 8.31
6c9cb4ce7750db02c578ebf3ae7a27f64a315289a4f9c75e2278b539db39f5c2  1953154411/FusionReactor.dll
7a0be59468323d021e141aa3e971fe3e6292c3781750b4e0a1ce150d667022f3  1953157233/Radiation.dll
82d9a827cb5a20c79b89fdae68a5ba75005efcc5f08e5ddca382bc5c3cceb3b9  1984954968/GigawattWire.dll
